shoes-swt/lib/shoes/swt/image.rb

```diff
@@ -77,10 +78,18 @@ def default_height
       end
 
       def download_and_display_real_image(url)
-        @tmpname = File.join(Dir.tmpdir, "__shoes4_#{Time.now.to_f}.png") unless @tmpname_or_data
+        if @is_image
```

Comment:
The repeated usage of `@is_image` suggests that polymorphism might be a better fit here, e.g. either an image subclass or another object we compose that takes care of this :)